packages/ai-jsx/src/lib/openai.tsx

```diff
@@ -321,7 +392,19 @@ export async function* OpenAIChatModel(
 
   yield AI.AppendOnlyStream;
 
-  const conversationMessages = await renderToConversation(props.children, render);
+  let promptTokenLimit = tokenLimitForChatModel(props.model, props.functionDefinitions);
+
+  // If maxTokens is set, reserve that many tokens for the reply.
```

Comment:
This is interesting. I feel like (not necessarily in this PR) we want a "minimum tokens reserved for reply" mechanism. e.g. in general, I don't want to limit the size of the response. But if my input is so large that I only have 100 tokens left in the window for the response, I want to shrink so I have more room.

Reply:
Yeah I was originally going to add that but scaled back for this PR because `maxTokens` is _almost_ that.

I could eventually imagine us doing something like detecting that the response ran out of tokens and then re-shrinking the conversation and re-invoking the model to continue on.